.github/workflows/build-and-run.yml

```diff
@@ -0,0 +1,109 @@
+name: Build-And-Run Tests
+
+on: [push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_and_test:
+    env:
+      ALL_EXCLUDE: stage_dem|ionosphere|background|pybind.unwrap.phass
```

Comment:
My thinking here was that we were concerned about the nature of these tests generally, i.e. that they were written in such a way that they had greater potential to fail across a variety of platforms and dependencies (that is to say, if we were to add more tests to this CI workflow, all of the tests here had either a greater potential to fail or were considered otherwise flawed)

With the exception of stage_dem of course, which simply has a permissions problem and will be tested elsewhere.

If you'd like, I can move them though!

Reply:
Oh I see the reasoning now, thanks.

It's a fair point, but I'm thinking that it'd be better to try and maximize coverage of each job, and only add a test to `ALL_EXCLUDE` if it's been demonstrated to fail across multiple platforms and/or sets of dependencies. It doesn't cost much to move a test to `ALL_EXCLUDE` in the future. 

It seems like the `background` test is failing on macOS because we made some assumptions about runtime that aren't satisfied on the slower GHA macOS runners, and the `ionosphere` tests seem to be failing in the Linux Minimum job due to some issue with older versions of GDAL (though I'm not sure exactly what). So, at least in those two cases, we have some mechanistic reason to believe the tests should fail only in specific jobs.